### PR TITLE
Update to 1.18 release

### PR DESCRIPTION
**Pull request by vanjmali**
Update Fabric dependencies and code for 1.18

### CODE REVIEW

**Comment by lukearnould** [comment]
How can I compile this pull request for myself? It seems that this project is abandoned. Or would there be a way you could compile it for me and offer a download?

**Comment by vanjmali** [comment]
I believe you just need to make sure to use Java 17.
If you are using IntelliJ IDEA you would need to set your project to use Java 17 by going to `File > Settings > Build, Execution, Deployment > Build Tools > Gradle` (or Gradle Settings) and set `Gradle JVM` to `17`.

That should be all, I think. It also requires to use of Gradle 7.2+ to support that Java version which should be done automatically.

Here's also compiled jar file of the mod (extract it):
[borderless-mining-1.1.0+1.18.jar.zip](https://github.com/comp500/BorderlessMining/files/7652622/borderless-mining-1.1.0%2B1.18.jar.zip)

**Comment by comp500** [comment]
> How can I compile this pull request for myself? It seems that this project is abandoned. Or would there be a way you could compile it for me and offer a download?

This project isn't abandoned! I'm just very busy with work and other projects, so I don't often have enough time to update them.

File: src/main/java/link/infra/borderlessmining/config/ConfigScreen.java [hunk 1]
```diff
@@ -122,6 +122,11 @@ public void tick() {
 		public Style getHoveredStyle(int mouseX, int mouseY) {
 			return null;
 		}
+
+//		@Override
+//		public List<? extends Selectable> selectableChildren() {
```
Comment (comp500) [inline src/main/java/link/infra/borderlessmining/config/ConfigScreen.java]: I assume this comment has no purpose, could you remove it?

**Review by comp500 — CHANGES_REQUESTED** [review]
Looks great, thanks! Just noted a small issue to fix.

**Comment by lukearnould** [comment]
> This project isn't abandoned! I'm just very busy with work and other projects, so I don't often have enough time to update them.

So sorry didn’t mean any disrespect! Can certainly understand life being too busy.

**Review by comp500 — APPROVED** [review]

**Comment by vanjmali** [comment]
I believe you should also update Github Workflow for 1.18 branch to use Java 17 when able to as it is not able to build it at the moment.

**Comment by comp500** [comment]
Already did https://github.com/comp500/BorderlessMining/commit/70af29256c45928b3ed20013b790c573706c7ccc :)

**Pull request merged**